src/vdom/diff.js

```diff
@@ -187,10 +187,10 @@ function innerDiffNode(dom, vchildren, context, mountAll, absorb) {
 		min = 0,
 		len = originalChildren.length,
 		childrenLen = 0,
-		vlen = vchildren && vchildren.length,
+		vlen = vchildren ? vchildren.length : 0,
 		j, c, vchild, child;
 
-	if (len) {
+	if (len!==0) {
```

Comment:
Strict equality should be faster than a comparison since it does no casting, but @bmeurer is far more knowledgeable on that than I am.